reporting/reporting.go

```diff
@@ -0,0 +1,127 @@
+package reporting
+
+import (
+	"context"
+	"encoding/json"
+	"strings"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/honeycombio/libhoney-go"
+
+	"github.com/honeycombio/honeytail/event"
+)
+
+const (
+	contextKeyBuilder = "reporting"
+	reportingSuffix   = "-reports"
+)
+
+func NewContext(ctx context.Context) context.Context {
+	builder := libhoney.NewBuilder()
```

Comment:
I feel like there should be some things shoved in to the builder here, but I'm not sure what. This tails into our discussion about this dataset as a grab bag of junk...
ooh! If you do tie in to the stats at all, a dynamic function representing the number of events processed since the last message would be kinda neat.

Reply:
Any particular reason you chose to put the dataset name munging in `getEvent` instead of here?

Reply:
Oops, nope, that was a bridge lost in the iteration swamp.

Reply:
>  a dynamic function representing the number of events processed since the last message would be kinda neat.

Hrmmmm. I hear what you're saying... except `AddDynamicField` gets called on each event we want to send, yeah? So wouldn't it just always be 1?

Is **this** machine's hostname useful or interesting? What about CPU utilization or i/o or something? Any votes for items there? :)

Reply:
> I had a similar issue and never found a lightweight implementation

[:'( via SO](https://stackoverflow.com/a/17783687)

And something like pprof probably isn't something we want to shove in by default. =\ Mrrr.

Reply:
Yes, I think this host's name is a good thing to add for when you're trying to diagnose anything running on a cluster. (I know at least a few folks use honeytail in this way.)